### PR TITLE
Add MySQL generated columns

### PR DESCRIPTION
**Pull request by hdogan**
Quick & dirty solution for #1276

This solution may be not necessary anymore if #1175 is accepted.

### CODE REVIEW

File: src/Phinx/Db/Table/Column.php [hunk 1]
```diff
@@ -121,6 +121,21 @@ class Column
      */
     protected $values;
 
+    /**
+     * @var boolean
```
Comment (dereuromark) [inline src/Phinx/Db/Table/Column.php]: bool

**Comment by dereuromark** [comment]
Looks OK to be merged after CS fixes.

**Comment by dereuromark** [comment]
@lorenzo Do you prefer https://github.com/cakephp/phinx/pull/1175 ? Or shall we continue with this?

**Comment by hdogan** [comment]
SQL Server has a similar featured called "computed column". See issue #1317 
I'm not exactly sure if #1175 solves those kind of problems. It would be nice to have some more examples of #1175 PR with generated/computed column scenario.

Or adapter class (or user) may inject custom/relaxed properties into column class then use it in own way. Like:

```php
// For MySQL generated columns
$table->addColumn('full_name', 'string', [
    'limit' => 100,
    'custom' => [
        'generated' => true,
        'stored' => true, // virtual if false or not defined
        'expression' => 'CONCAT(first_name, \' \', last_name)'
    ]
]);

// For SQL Server computed columns
$table->addColumn('full_name', 'string', [
    'limit' => 100,
    'custom' => [
        'computed' => true,
        'persisted' => true, // not persisted if false or not defined
        'expression' => 'CONCAT(first_name, \' \', last_name)'
    ]
]);
```

**Comment by dereuromark** [comment]
The other PR was merged. I guess we can close?

**Comment by designermonkey** [comment]
@dereuromark you've stated that 'the other pr was merged' but I'm not sure what you're referencing?

I have come across this looking for support for generated columns and I can't find anything about it in the codebase. Is there support added or not? If so how do I use it?

/cc @hdogan

**Comment by dereuromark** [comment]
Wasnt it https://github.com/cakephp/phinx/pull/1175 ?

**Comment by designermonkey** [comment]
How would that be used to create generated columns? Is the literal value just used as is? Some examples other than simple types might be useful in the docs, as I completely missed that this could be used for generated columns :/

**Comment by dereuromark** [comment]
All right, maybe you can reopen a PR with the suggested changes and we can try to fit it in asap?

**Comment by designermonkey** [comment]
I wouldn't know how to document this feature, that's why I'm looking for it and failing to spot how to do it.

If I could, I would.

**Pull request closed without merging**